tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -958,39 +959,64 @@ static inline struct mount *real_mount(struct vfsmount *mnt)
     return container_of(mnt, struct mount, mnt);
 }
 
-static __always_inline u32 get_inet_rcv_saddr(struct inet_sock *inet)
+static __always_inline u16 get_inet_sport(struct inet_sock *inet)
 {
-    return READ_KERN(inet->inet_rcv_saddr);
+    return READ_KERN(inet->inet_sport);
 }
 
-static __always_inline u32 get_inet_saddr(struct inet_sock *inet)
+static __always_inline struct sock* get_socket_sock(struct socket *socket)
 {
-    return READ_KERN(inet->inet_saddr);
+    return READ_KERN(socket->sk);
 }
 
-static __always_inline u32 get_inet_daddr(struct inet_sock *inet)
+static __always_inline struct sock_common * get_sock_sk_common(struct sock *sock)
```

Comment:
Better to avoid reading the whole sock_common struct (#1193) + I don't see a reason to use a map here.

Reply:
the reason for the bug is using:
`READ_KERN(sk->__sk_common.skc_dport)`

it's just doesn't get the right field. (i've noticed it gets the skc_family field intead, which is 32 bits below)


using:
```
struct sock_common sk_common = READ_KERN(sk->__sk_common);
return sk_common.skc_dport;
```

solves it.

Reply:
but then we reach the BPF stack limit, so struct sock_common needs to be saved in a map.

Reply:
> the reason for the bug is using: `READ_KERN(sk->__sk_common.skc_dport)`
> 
> it's just doesn't get the right field. (i've noticed it gets the skc_family field intead, which is 32 bits below)
> 
> using:
> 
> ```
> struct sock_common sk_common = READ_KERN(sk->__sk_common);
> return sk_common.skc_dport;
> ```
> 
> solves it.

You can read a pointer to sock_common instead of the whole struct, then use it to read the specific field